src/hu_nmt/data_augmentator/entrypoints/precompute_paralell_dependency_trees.py

```diff
@@ -0,0 +1,57 @@
+import os
+import click
+from typing import Iterator
+
+from hu_nmt.data_augmentator.dependency_parsers.spacy_dependency_parser import SpacyDependencyParser
+from hu_nmt.data_augmentator.utils.logger import get_logger
+from hu_nmt.data_augmentator.dependency_parsers.english_dependency_parser import EnglishDependencyParser
+
+log = get_logger(__name__)
+
+
+@click.command()
+@click.argument('src_tag')
+@click.argument('tgt_tag')
+@click.argument('src_input_path')
+@click.argument('tgt_input_path')
+@click.argument('dep_tree_output_path')
+@click.argument('file_batch_size')
+def main(src_tag, tgt_tag, src_input_path, tgt_input_path, dep_tree_output_path, file_batch_size):
+    dep_parsers = {
+        'en': EnglishDependencyParser(),
+        'hu': SpacyDependencyParser(lang='hu')
+    }
```

Comment:
they are command line parameters, maybe I can make a global dictionary for choosing parsers with the language tag

Reply:
Maybe a more elegant solution would be to create a `DependencyParserFactory` class, which has a method that takes a language code and returns a dependency parser instance for that language. But I can live with dictionaries as well if you think this is an unnecessarry abstraction.

Reply:
I'd also advise some renaming in the script:
- `src_tag` --> `src_lang_code`
- `tgt_tag` --> `tgt_lang_code`
"Tag" I find a bit confusing, because we expect these variables in the form of ISO 639-1 language codes